### DISCUSSION TITLE
Sql Injection using py_sql derive macro

### DISCUSSION THREAD

**Discussion by nanokes** [rbatis/rbatis#272] [General]
I was working with the py_sql. I am new to this crate so please if I am doing something wrong I am sorry, but I found it quite easy to make a working Sql Injection attack.

_Using the code below_
```rust
#[py_sql(
  RB,
  "SELECT * FROM `users` WHERE `${column}` = '${value}';"
)]
pub async fn find_user(column: &str, value: &str) -> Vec<UserAccount> {}
```

If I use `person@mail.com' OR '1' = '1` as `value`, then instead of returning all rows that meet a certain given requirement, it just returns every single row in the database. The SQL equivalent of this would be (if `column = "email"`)
```sql
SELECT * FROM `users` WHERE `email` = 'rowan@gmal.com' OR '1' = '1';
```
One hopefully simple fix for this would be to just take every quote inputted into `value` and turn it into its ASCII Code equivalent. I have worked a lot with Objectionjs in the past in JavaScript and this is more-or-less how they handle attack vectors like this.

**Comment by zhuxiujia** [comment]
> 

If you need a safe parameter, use #{value}. This will be precompiled by the database to prevent SQL injection。

and，just like ``` #{1+1}，#{'a'+'b'}  ```Many expressions are allowed

**Comment by nanokes** [comment]
@zhuxiujia Thanks for the reply! You were lots of help.
